### PR TITLE
Adding module for miniprot_index. Closes #1.

### PR DESCRIPTION
**Pull request by yumisims**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/miniprot/index/meta.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      type: file
+      description: |
+        Reference database in FASTA format.
+output:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - mmi:
```
Comment (priyanka-surana) [inline modules/miniprot/index/meta.yml:30]: name here should match the name in `main.nf` -- use `index` here as well.

File: tests/modules/miniprot/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { MINIPROT_INDEX } from '../../../../modules/miniprot/index/main.nf'
+
+workflow test_miniprot_index {
+
+    /*fasta = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)*/
+    fasta =file('https://tolit.cog.sanger.ac.uk/test-data/Oscheius_sp/assembly/nxOscSpef2.fasta', checkIfExists: true)
```
Comment (priyanka-surana) [inline tests/modules/miniprot/index/main.nf:10]: Can you check if the data is synced? Also check the test runs successfully locally, just to be sure.

To create `test.yml` – `nf-core modules create-test-yml`
To test – `nf-core modules test`

Comment (yumisims) [reply tests/modules/miniprot/index/main.nf:10]: the data has been synced

File: modules/miniprot/index/main.nf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    """
+    miniprot \\
+        -t $task.cpus \\
+        -d ${fasta.baseName}.mpi \\
+        $args \\
+        $fasta
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        miniprot: \$(miniprot --version 2>&1)
```
Comment (gq1) [inline modules/miniprot/index/main.nf:31]: no --version flag for miniprot

File: modules/miniprot/index/main.nf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    miniprot \\
+        -t $task.cpus \\
+        -d ${fasta.baseName}.mpi \\
+        $args \\
+        $fasta
+
```
Comment (priyanka-surana) [inline modules/miniprot/index/main.nf:28]: There is whitespace here that is causing Editor Config linting error.

**Comment by gq1** [comment]
@priyanka-surana do you have any idea for the linting error? Can you check please?

**Comment by priyanka-surana** [comment]
> @priyanka-surana do you have any idea for the linting error? Can you check please?

Might have something to do with this: https://github.com/sanger-tol/nf-core-modules/blob/b7c199ab374754fb3ee3eca7dd09089b70e6314d/.github/workflows/nf-core-linting.yml#L57

File: modules/miniprot/index/meta.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      type: file
+      description: |
+        Reference database in FASTA format.
+output:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - mgi:
```
Comment (priyanka-surana) [inline modules/miniprot/index/meta.yml:30]: This needs to be `index` to match the `main.nf`

**Review by priyanka-surana — CHANGES_REQUESTED** [review]

**Comment by yumisims** [comment]
> 

should we just ignore the lint error for now then? @priyanka-surana

**Comment by priyanka-surana** [comment]
> should we just ignore the lint error for now then? @priyanka-surana

Can you do `nf-core modules lint` locally and make sure it passes

**Comment by yumisims** [comment]
@priyanka-surana local int success

**Review by priyanka-surana — APPROVED** [review]
LGTM - `nf-core lint` works locally for @yumisims

**Pull request merged**